pkg/actions/actions.go

```diff
@@ -0,0 +1,291 @@
+	ChartsScriptOptionsFile = "configuration.yaml"
+)
+
+func List(currentPackage string, porcelainMode bool) {
+	repoRoot := getRepoRoot()
+	packageList, err := charts.ListPackages(repoRoot, currentPackage)
+	if err != nil {
+		logrus.Fatal(err)
+	}
+	if porcelainMode {
+		fmt.Println(strings.Join(packageList, " "))
+		return
+
+	}
+	logrus.Infof("Found the following packages: %v", packageList)
+}
+
+func Prepare(currentPackage string) {
+	packages := getPackages(currentPackage)
+	if len(packages) == 0 {
+		logrus.Fatal("Could not find any packages in packages/")
+	}
+	for _, p := range packages {
+		if err := p.Prepare(); err != nil {
+			logrus.Fatal(err)
+		}
+	}
+}
+
+func Patch(currentPackage string) {
+	packages := getPackages(currentPackage)
+	if len(packages) == 0 {
+		logrus.Infof("No packages found.")
+		return
+	}
+	if len(packages) != 1 {
+		packageNames := make([]string, len(packages))
+		for i, pkg := range packages {
+			packageNames[i] = pkg.Name
+		}
+		logrus.Fatalf(
+			"PACKAGE=\"%s\" must be set to point to exactly one package. Currently found the following packages: %s",
+			currentPackage, packageNames,
+		)
+	}
+	if err := packages[0].GeneratePatch(); err != nil {
+		logrus.Fatal(err)
+	}
+}
+
+func Charts(currentPackage string) {
+	packages := getPackages(currentPackage)
+	if len(packages) == 0 {
+		logrus.Infof("No packages found.")
+		return
+	}
+	chartsScriptOptions := parseScriptOptions()
+	for _, p := range packages {
+		if err := p.GenerateCharts(chartsScriptOptions.OmitBuildMetadataOnExport); err != nil {
+			logrus.Fatal(err)
+		}
+	}
+}
+
+func Index() {
+	repoRoot := getRepoRoot()
+	if err := helm.CreateOrUpdateHelmIndex(filesystem.GetFilesystem(repoRoot)); err != nil {
+		logrus.Fatal(err)
+	}
+}
+
+func Zip(currentChart string) {
+	repoRoot := getRepoRoot()
+	if err := zip.ArchiveCharts(repoRoot, currentChart); err != nil {
+		logrus.Fatal(err)
+	}
+	Index()
+}
+
+func Unzip(currentAsset string) {
+	repoRoot := getRepoRoot()
+	if err := zip.DumpAssets(repoRoot, currentAsset); err != nil {
+		logrus.Fatal(err)
+	}
+	Index()
+}
+
+func Clean(currentPackage string) {
+	packages := getPackages(currentPackage)
+	if len(packages) == 0 {
+		logrus.Infof("No packages found.")
+		return
+	}
+	for _, p := range packages {
+		if err := p.Clean(); err != nil {
+			logrus.Fatal(err)
+		}
+	}
+}
+
+func Validate() {
+	chartsScriptOptions := parseScriptOptions()
+
+	logrus.Infof("Checking if Git is clean")
+	checkGit()
+
+	logrus.Infof("Generating charts")
+	Charts("")
+
+	logrus.Infof("Checking if Git is clean after generating charts")
+	checkGit()
+	logrus.Infof("Successfully validated that current charts and assets are up to date.")
+
+	pullUpstream(chartsScriptOptions)
+
+	logrus.Info("Zipping charts to ensure that contents of assets, charts, and index.yaml are in sync.")
+	Zip("")
+
+	logrus.Info("Doing a final check to ensure Git is clean")
+	checkGit()
+
+	logrus.Info("Successfully validated current repository!")
```

Comment:
I think this should just call ValidateLocal and ValidateRemote and we should modify the log messages accordingly to make it look good.

I also think this should take in `remote` and `local` bools as function arguments and do the checking to see that exactly 1 of them evaluates to true within this function rather than in the main.go.

Reply:
In the current flow if you call it with both local and remote false the logic is to do both local and remove checking.  This makes sense with a CLI, however with mage if you call `mage validate false false` to me it would seem somewhat counterintuitive.  That's why I split them into multiple functions so a mage user could call `mage validatelocal` or `mage validateremote` to call remote or local only.

One better way might be to have constant string types e.g.
```
type ValidateType string
const (
  ValidateAll ValidateType = ""
  ValidateLocal ValidateType = "local"
  ValidateRemote ValidateType = "remote"
)
```
Because of the restrictions in mage the validate method would need to accept a string argument, but this might be a cleaner way to deal with it?